File: estudosJava/CursoJavaCompletoUdemy/ExerciciosFixacao/Ex006V2/src/main/java/entities/DadosConta.java
package entities;

public class DadosConta {
    private int numeroConta;
    private String titular;
    private double saldo;

    public DadosConta(int numeroConta, String titular, double depositoInicial) {
        this.numeroConta = numeroConta;
        this.titular = titular;
        depositoSaldo(depositoInicial);
    }

    public DadosConta(int numeroConta, String titular) {
        this.numeroConta = numeroConta;
        this.titular = titular;
    }

    public void depositoSaldo(double valor){
        this.saldo += valor;
    }
    public  void retiradaSaldo(double valor){
        this.saldo -= valor + 5.00; //5.00 é a taxa do saque.
    }

    public String toString() {
        return "Número da Conta: "
                + numeroConta
                + ", Titular: "
                + titular
                + ", Saldo R$ "
                + String.format("%.2f", saldo);
    }

    public int getNumeroConta() {
        return numeroConta;
    }

    public String getTitular() {
        return titular;
    }

    public void setTitular(String titular) {
        this.titular = titular;
    }

    public double getSaldo() {
        return saldo;
    }
}

File: C#/UDEMY_CSharp_Completo_POO/_060_ExercicioFixacaoPOO/_060_ExercicioFixacaoPOO/DadosConta.cs
using System.Globalization;
namespace _060_ExercicioFixacaoPOO
{
    class DadosConta
    {
        private string _nome;
        public int Conta { get; private set; }
        public double Saldo { get; private set; }

        public DadosConta(string nome, int conta)
        {
            _nome = nome;
            Conta = conta;
        }
        public DadosConta(string nome, int conta, double depositoInicial) : this(nome, conta)
        {
            Deposito(depositoInicial);
        }


        public string Nome
        {
            get { return _nome; }
            set
            {
                if (value != null && value.Length > 1)
                {
                    _nome = value;
                }
            }
        }

        public void Deposito(double valor)
        {
            Saldo += valor;
        }

        public void Saque(double valor)
        {
            Saldo -= valor + 5.00;
        }

        public override string ToString()
        {
            return "Conta "
                + Conta
                + ", Titular: "
                + Nome
                + ", Saldo: $ "
                + Saldo.ToString("F2", CultureInfo.InvariantCulture);
        }
    }
}